src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs

```diff
@@ -91,6 +91,11 @@ public void Format(LogEvent logEvent, TextWriter output)
 
                     token.Render(overridden, output, _formatProvider);
                 }
+                else if (OutputProperties.IsLevelProperty(pt))
```

Comment:
Using `is`/`as` will be more efficient than the name check and subsequent cast here, and gets rid of the `IsLevelProperty` method. E.g.:

```
else
{
    var lp = propertyValue as LogEventLevelValue;
    if (lp != null)
    {
        lp.Render(...)
```